examples/filetable.py

```diff
@@ -15,80 +16,96 @@ def filetable_to_firefly(ffclient, topdir, pattern, recursive=True):
-                                     suffix='.csv') as fd:
+    filelist = sorted(glob(topdir + "/**/" + pattern, recursive=recursive))
+    metadict = {"datasource": "path"}
+    with tempfile.NamedTemporaryFile(mode="w+t", delete=False, suffix=".csv") as fd:
         csv_writer = csv.writer(fd)
-        csv_writer.writerow(['number','name','path'])
+        csv_writer.writerow(["number", "name", "path"])
         for i, path in enumerate(filelist):
-            csv_writer.writerow([i, os.path.basename(path),
-                                 'file://'+path])
+            # Docker Firefly allows uploads from /external
+            csv_writer.writerow([i,
+                                 os.path.basename(path),
+                                 "file:///external" + path])
 
     tbl_val = ffclient.upload_file(fd.name)
     os.remove(fd.name)
-    return(tbl_val, metadict)
+    return (tbl_val, metadict)
 
 
 # Sample application
 def main():
     import argparse
-    parser = argparse.ArgumentParser(description="""
+
+    parser = argparse.ArgumentParser(
+        description="""
         Display a table of files in a Firefly window
-        """)
-    parser.add_argument('topdir', help='top-level directory to search')
-    parser.add_argument('pattern', help='filename pattern for search')
-    parser.add_argument('--norecursion', help='do not recursively search topdir',
-                        action='store_true')
-    parser.add_argument('--host', help='host address for Firefly',
-                        default='localhost')
-    parser.add_argument('--port', help='port for Firefly', default='8080')
-    parser.add_argument('--base', help='base.for Firefly', default='firefly')
-    parser.add_argument('--channel', help='channel name for websocket',
-                        default=None)
-    parser.add_argument('--printurl', help='print browser url instead of' +
-                        ' attempting to launch browser', action='store_true')
+        """
+    )
+    parser.add_argument("topdir", help="top-level directory to search")
+    parser.add_argument("pattern", help="filename pattern for search")
+    parser.add_argument(
+        "--norecursion", help="do not recursively search topdir", action="store_true"
+    )
+    parser.add_argument(
+        "--firefly_url", help="URL for Firefly server", default=os.getenv("FIREFLY_URL")
+    )
+    parser.add_argument("--channel", help="channel name for websocket", default=None)
+    parser.add_argument(
+        "--printurl",
+        help="print browser url instead of" + " attempting to launch browser",
+        action="store_true",
+    )
 
     args = parser.parse_args()
     topdir = args.topdir
     pattern = args.pattern
-    host = args.host
-    port = args.port
-    basedir = args.base
+    firefly_url = args.firefly_url
     channel = args.channel
     printurl = args.printurl
+    launch_browser = True
+    if printurl:
+        launch_browser = False
```

Comment:
[Optional] can be simplified:
```suggestion
    launch_browser = False if printurl else True
```